python/lsst/cp/verify/verifyBfk.py

```diff
@@ -0,0 +1,237 @@
+        doc="Output statistics from cp_verify.",
+        storageClass="StructuredDataDict",
+        dimensions=["instrument", "visit", "detector"],
+    )
+
+
+class CpVerifyBfkConfig(CpVerifyStatsConfig,
+                        pipelineConnections=CpVerifyBfkConnections):
+    """Inherits from base CpVerifyStatsConfig.
+    """
+
+    matchRadiusPix = pexConfig.Field(
+        dtype=float,
+        default=3,
+        doc=("Match radius for matching icSourceCat objects to sourceCat objects (pixels)"),
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.catalogStatKeywords = {'BRIGHT_SLOPE': None,
+                                    'NUM_MATCHES': None,
+                                   }  # noqa F841
+
+
+def exponentialModel(x, A, alpha, x0):
+    """An exponential model.
+    """
+    return A * np.exp(alpha * (x - x0))
+
+
+def modelResidual(p, x, y):
+    """Model residual for fit below.
+    """
+    return y - exponentialModel(x, *p)
+
+
+class CpVerifyBfkTask(CpVerifyStatsTask):
+    """Bfk verification sub-class, implementing the verify method.
+    """
+
+    ConfigClass = CpVerifyBfkConfig
+    _DefaultName = 'cpVerifyBfk'
+
+    def catalogStatistics(self, exposure, catalog, uncorrectedCatalog, statControl):
+        """Measure the catalog statistics.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            The exposure to measure.
+        catalog : `lsst.afw.table.Table`
+            The catalog to measure.
+        uncorrectedCatalog : `lsst.afw.table.Table`
+            The uncorrected catalog to measure.
+        statControl : `lsst.afw.math.StatisticsControl`
+            Statistics control object with parameters defined by
+            the config.
+
+        Returns
+        -------
+        outputStatistics : `dict` [`str`, `dict` [`str`, scalar]]
+            A dictionary indexed by the amplifier name, containing
+            dictionaries of the statistics measured and their values.
+        """
+        outputStatistics = {}
+
+        matches = afwTable.matchXy(catalog, uncorrectedCatalog, self.config.matchRadiusPix)
+        outputStatistics['NUM_MATCHES'] = len(matches)
+
+        magnitude = []
+        sizeDiff = []
+        for source, uncorrectedSource, d in matches:
+            # This uses the simple difference in source moments.
+            sourceMagnitude = -2.5 * np.log10(source.getPsfInstFlux())
+            sourceSize = source['base_SdssShape_xx'] + source['base_SdssShape_yy']
+            uncorrectedSize = uncorrectedSource['base_SdssShape_xx'] + uncorrectedSource['base_SdssShape_yy']
+
+            magnitude.append(float(sourceMagnitude))
+            sizeDiff.append(float(uncorrectedSize - sourceSize))
+
+        mask = np.isfinite(magnitude) * np.isfinite(sizeDiff)
+        if 'BRIGHT_SLOPE' in self.config.catalogStatKeywords:
+            exponentialFit = least_squares(modelResidual, [0.0, 0.0, 0.0],
```

Comment:
Exactly zero is not always the most robust starting value to use, especially if, for example, you know it's always going to be positive. Even just bumping to 0.1 might help for some cases.

Reply:
I've made the default `0.01` to avoid any issues.